pkg/apis/provisioning/v1alpha5/constraints.go

```diff
@@ -54,7 +55,11 @@ func (c *Constraints) ValidatePod(pod *v1.Pod) error {
 	// Test if pod requirements are valid
 	requirements := NewPodRequirements(pod)
 	if errs := requirements.Validate(); errs != nil {
-		return fmt.Errorf("pod requirements not feasible, %v", errs)
+		var multiErrors error
+		for _, err := range errs {
+			multiErrors = multierr.Append(multiErrors, fmt.Errorf(err))
```

Comment:
IMO this function shouldn't be aware of the multierr. Encapsulate this inside requirements.

Reply:
I was thinking it may be nice to allow functions that raise errors to determine which message among all the error messages should be reported at runtime.
This is the same model that the runtime validation API is using (see `validation.IsQualifiedName(key)`).

Reply:
done